sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -1615,9 +1615,7 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog with QueryErrorsB
             case (CharType(l1), CharType(l2)) => l1 == l2
             case (CharType(l1), VarcharType(l2)) => l1 <= l2
             case (VarcharType(l1), VarcharType(l2)) => l1 <= l2
-            case _ =>
-              Cast.canUpCast(from, to) ||
-                DataType.equalsIgnoreCompatibleCollation(field.dataType, newDataType)
```

Comment:
why is this change?

Reply:
seems a reasonable cleanup but unrelated to this PR, right?

Reply:
DataType.equalsIgnoreCompatibleCollation(field.dataType, newDataType) is redundant, if it returns true then so does Cast.canUpCast(from, to)